test/utils.jl

```diff
@@ -21,9 +25,11 @@ function generate_dataset(
     model, optimizer; N::Integer, dim_x::Integer, dim_y::Integer, σ::Real
 )
     X = [randn(dim_x) for n in 1:N]
-    thetas = [model(X[n]) + σ * randn(dim_y) for n in 1:N]
-    Y = [optimizer(θ) for θ in thetas]
-    return (X=X, thetas=thetas, Y=Y)
+    thetas = [model(x) for x in X]
+    noiseless_Y = [optimizer(θ) for θ in thetas]
+    noisy_thetas = [θ + σ * randn(dim_y) for θ in thetas]
+    noisy_Y = [optimizer(θ) for θ in noisy_thetas]
+    return (X=X, thetas=thetas, noisy_Y=noisy_Y, noiseless_Y=noiseless_Y)
```

Comment:
Not sure we need to generate both: I actually meant to use `σ = 0` to generate the traning set and `σ > 0` to generate the test set. Is that possible?

Reply:
Solved by @BatyLeo - we actually need a noiseless Y_train to compute the objective gap